File: baekjoon/3_의_배수.js
let input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("")
  .map(Number);

let count = 0;

while (input.length > 1) {
  const sum = input.reduce((acc, cur) => acc + cur);
  input = sum.toString().split("").map(Number);
  count++;
}
console.log(count);
console.log(input[0] % 3 === 0 ? "YES" : "NO");

File: baekjoon/3_의_배수.rs
fn main() {
    let mut chars = get_chars_from_stdin();
    let mut count = 0;
    loop {
        if chars.len() == 1 {
            break;
        }
        let mut sum = 0;
        for ch in chars.iter() {
            sum += *ch as i32 - 48;
        }
        count += 1;
        chars = sum.to_string().bytes().collect();
    }
    println!("{}", count);
    if chars[0] % 3 == 0 {
        println!("YES");
    } else {
        println!("NO");
    }
}

fn get_chars_from_stdin() -> Vec<u8> {
    let mut st = String::new();
    std::io::stdin().read_line(&mut st).unwrap();
    let ret = st.trim().bytes().collect();

    ret
}
